src/cpu/aarch64/acl_gemm_convolution.cpp

```diff
@@ -14,30 +14,142 @@
 * limitations under the License.
 *******************************************************************************/
 
-#include "cpu/aarch64/acl_gemm_convolution.hpp"
+#include "acl_gemm_convolution.hpp"
+#include "acl_convolution_utils.hpp"
+#include "common/memory_tracking.hpp"
 
 namespace dnnl {
 namespace impl {
 namespace cpu {
 namespace aarch64 {
 
-// src_type is enum and is mapped to src_data_t via
-// prec_traits<src_type>::type src_data_t
-template <data_type_t src_type, data_type_t wei_type, data_type_t dst_type,
-        data_type_t bia_type>
-status_t acl_gemm_convolution_fwd_t<src_type, wei_type, dst_type,
-        bia_type>::execute_forward(const exec_ctx_t &ctx) const {
-    // Lock here is needed because resource_mapper does not support
-    // concurrent multithreaded access.
-    std::lock_guard<std::mutex> _lock {this->mtx};
-    // Retrieve primitive resource and configured Compute Library objects
-    auto *acl_resource = ctx.get_resource_mapper()->get<acl_resource_t>(this);
-    acl_obj_t<arm_compute::NEGEMMConvolutionLayer> &acl_obj
-            = acl_resource->get_acl_obj();
-
-    return execute_forward_conv_acl<
-            acl_obj_t<arm_compute::NEGEMMConvolutionLayer>, pd_t, src_data_t,
-            wei_data_t, dst_data_t, bia_data_t>(ctx, acl_obj, pd());
+namespace {
+// Keys are anonymous. So deduce the type automagically.
+using conv_key_t = decltype(memory_tracking::names::key_gemm_tmp_buffer);
+
+// Map: [slot , key]
+const std::map<int, conv_key_t> gemm_conv_keys
```

Comment:
It looks like this could be an array, I'm guessing it's a map so that it mirrors the ACL aux mem datastructure? If so, is it worth a short comment to say that these ints have meaning?

Reply:
Added comments for clarification.

Reply:
Given that those are related to arm_compute::experimental::MemoryRequirements ids, would it make sense to expose those indices as an enum type in ACL?

Reply:
BTW, when looking at init_scratchpad function, it seems that it iterates over keys defined by the oneDNN impl. 
Wouldn't it be more robust if  arm_compute::experimental::MemoryRequirements provided a way to iterate over ids of required memory?